src/document_basic.c

```diff
@@ -173,14 +191,16 @@ int Document_LoadSchemaFieldJson(Document *doc, RedisSearchCtx *sctx) {
     FieldSpec *field = &spec->fields[ii];
 
     jsonIter = japi->get(jsonRoot, field->path);
+    // if field does not exist or is empty (can happen after JSON.DEL)
     if (!jsonIter) {
         continue;
     }
 
-    RedisJSON jsonValue = japi->next(jsonIter);
-    if (!jsonValue || 
-        FieldSpec_CheckJsonType(field->types, japi->getType(jsonValue)) != REDISMODULE_OK) {
-      goto done;
+    size_t len = japi->len(jsonIter);
+    if (len == 0) { // Q: Should we fail or assert?
```

Comment:
No, you might get an iterator with no results

Reply:
How come an array may not have any items?

Reply:
It's not an array, it's an iterator of results, you might give a json path that returns no results...